mitxgraders/baseclasses.py

```diff
@@ -26,6 +26,10 @@ class InvalidInput(Exception):
     """Raised whenever user input is invalid"""
     pass
 
+class StudentFacingError(Exception):
```

Comment:
> I think InvalidInput has a huge amount of overlap with StudentFacingError. I think having a base class of exceptions that go straight through is a great idea though, and we should refactor InvalidInput appropriately. One possible catch: if such an error is raised on an instructor's input, do we need to recast it as a ConfigError?

That was my inclination too. I think we should do that refactoring, but it might be best to do in a subsequent PR—seems like I might end up changing a bunch of files that aren't really related to vectors/matrices. But let me know if you feel otherwise.

Reply:
I agree - let's do that separately.